backend/trading/monitoring/learning.py

```
@@ -0,0 +1,484 @@
+        
+        # Training state
+        self.last_retrain: Optional[datetime] = None
+        self.samples_since_retrain: int = 0
+        self.retrain_history: List[Dict] = []
+        
+        # Validation results
+        self.validation_results: List[Dict] = []
+    
+    def add_sample(
+        self,
+        features: Dict[str, float],
```

Ruff (0.14.8)

Unused method argument: `features`

(ARG002)